float32 full
float32 left_low
float32 right_low
